collect_app/src/androidTest/java/org/odk/collect/android/FormNavigationTestCase.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android;
+
+import android.content.res.AssetManager;
+import android.os.Environment;
+import android.support.test.InstrumentationRegistry;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.odk.collect.android.listeners.FormLoaderListener;
+import org.odk.collect.android.logic.FormController;
+import org.odk.collect.android.tasks.FormLoaderTask;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+
+import timber.log.Timber;
+
+import static junit.framework.Assert.assertEquals;
+
+/**
+ * This test has been created in order to check indices while navigating through a form.
+ * It's especially important while navigating through a form that contains nested groups and if we
+ * use groups with field-list appearance because in that case we need to collect all indices of
+ * questions we want to display on one page (we need to recursively get all indices contained in
+ * such a group and its children). It might be also tricky when navigating backwards because then we
+ * need to navigate to an index of the first question of all we want to display on one page.
+ */
+@RunWith(Parameterized.class)
+public class FormNavigationTestCase {
+
+    private static final String FORMS_DIRECTORY = "/odk/forms/";
+
+    @Parameters(name = "{0}")
+    public static Iterable<Object[]> data() {
+        // Expected indices when swiping forward until the end of the form and back once.
+        // An index of -1 indicates the start or end of a form
```

Comment:
A single comment with no period is fine. But when there are two sentences together they should both end with periods.